[UTC] Record TBAA semantics when autogenerating check lines

antoniofrighetto opened the pull request:
UpdateTestChecks have been updated to take into account TBAA semantics as well, when emitting checks. This is achieved by parsing TBAA metadata for each tool invocation – whose tool is identified by their prefixes –, and maintaining a global dict of prefixes, TBAA nodes.

nikic reviewed (commented):
Looks like a reasonable feature, but someone with more Python skills should probably review this...

I'd suggest running this over all tests that have tbaa metadata to check whether any of the assertions fire.

antoniofrighetto commented:
> I'd suggest running this over all tests that have tbaa metadata to check whether any of the assertions fire.

Assuming this is to be done locally, or should we take the opportunity to update all UTC (possibly just version 5) tests that have TBAA metadata, here or separately?

nikic commented:
> > I'd suggest running this over all tests that have tbaa metadata to check whether any of the assertions fire.
> 
> Assuming this is to be done locally, or should we take the opportunity to update all UTC (possibly just version 5) tests that have TBAA metadata, here or separately?

Just locally for now. But updating tests in a followup may also make sense.

dtcxzyw commented:
> Hit an assert in a few tests that are using the `new-struct-path-tbaa` option, while testing locally. Relaxed the assert and left a TODO for now, as I couldn't find description of the new TBAA format in the LangRef, and I'm not sure if adoption to the new format is still an open work.

Hopefully the following links will help:
https://github.com/llvm/llvm-project/commit/04e1d01736d3368fd491d4bdb99a52452e9e83e1
https://lists.llvm.org/pipermail/llvm-dev/2017-November/118748.html

rjmccall commented:
I have completely forgotten everything about this, sorry, and I'm not familiar with this tool. You're asking me to review the PRs that update tests to use the new TBAA format?

antoniofrighetto commented:
> I have completely forgotten everything about this, sorry, and I'm not familiar with this tool. You're asking me to review the PRs that update tests to use the new TBAA format?

No problem at all. I was wondering if you could suggest somebody else who might take another look at this. Though, perhaps first, do the Clang tests update align with what you had originally in mind?

efriedma-quic reviewed (approved):
LGTM

I think the new variable names look fine.  And I won't claim to be a Python expert, but I don't see any obvious problems with the implementation.

antoniofrighetto committed:
[UTC] Introduce test for PR147670 (NFC)

antoniofrighetto committed:
[UTC] Record TBAA semantics when autogenerating check lines

UpdateTestChecks have been updated to take into account TBAA
semantics as well, when emitting checks. This is achieved by
parsing TBAA metadata for each tool invocation – whose tool
is identified by their prefixes –, and maintaining a global
dict of prefixes, TBAA nodes.
